### DISCUSSION TITLE
I want to create sound effect like hall, Big room, etc for use in am using Echo (by changing its parameters ) but it gives error continuously. Please help me

### DISCUSSION THREAD

**Discussion by SandeepJagdev** [pschatzmann/arduino-audio-tools#1008] [Q&A] [answered]
here is my code:
/**
 * @file basic-a2dp-audiokit.ino
 * @brief see https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-audiokit/basic-a2dp-audiokit/README.md
 * 
 * @author Phil Schatzmann
 * @copyright GPLv3
 */

#include "AudioTools.h"
#include "AudioLibs/AudioKit.h"
#include "AudioLibs/AudioA2DP.h"

BluetoothA2DPSink a2dp_sink;
AudioKitStream kit;
const int sample_rate = 44100;
const int channels = 1;
const int bits_per_sample = 16;

  uint16_t duration_ms = 1000;
  float depth = 0.5;
  float feedbackAmount = 0.5;
  uint32_t sampleRate = 44100;
    bool zeroIfBufferEmpty = false;

// Effects
Delay dly(duration_ms, depth,feedbackAmount,sampleRate,zeroIfBufferEmpty);
// Audio
//AudioKitStream kit; // Access I2S as stream
AudioEffects<GeneratorFromStream<effect_t>> effects(kit, channels , 1.0); // apply effects on kit
GeneratedSoundStream<int16_t> in(effects);
StreamCopy copier(kit, in); // copy in to kit

// Write data to AudioKit in callback
void read_data_stream(const uint8_t *data, uint32_t length) {
    in.write(data, length);
}

void setup() {
  Serial.begin(115200);
  AudioLogger::instance().begin(Serial, AudioLogger::Info);

  // setup output
  auto cfg = kit.defaultConfig(TX_MODE);
  cfg.sd_active = false;
  kit.begin(cfg);
dly.setActive(1);
 effects.addEffect(dly);

  // register callback
  a2dp_sink.set_stream_reader(read_data_stream, false);
  a2dp_sink.start("AudioKit");
}

void loop() {
 // kit.processActions();
   copier.copy();
}

**Comment by pschatzmann** [comment, marked as answer]
I suggest you build your output chain using an AudioEffectStream and then just write to it in the callback.

**Reply by SandeepJagdev** [reply]
any example??

**Comment by SandeepJagdev** [comment]
Thanks a lot . now it is working after two days of work.
